src/libsystemd-network/lldp-neighbor.c

```diff
@@ -776,7 +776,7 @@ int lldp_neighbor_build_json(sd_lldp_neighbor *n, JsonVariant **ret) {
         (void) sd_lldp_neighbor_get_system_name(n, &system_name);
         (void) sd_lldp_neighbor_get_system_description(n, &system_description);
 
-        valid_cc = sd_lldp_neighbor_get_enabled_capabilities(n, &cc);
+        valid_cc = sd_lldp_neighbor_get_enabled_capabilities(n, &cc) >= 0;
```

Comment:
Looking at "opbwrtdacsm " it seems CAPS lit up with this patch applied:
```sh
LINK  SYSTEM-NAME SYSTEM-DESCRIPTION CHASSIS-ID                       PORT-ID    PORT-DESCRIPTION CAPS
host0 H           -                  c9b646d89e8c45d5b5d3e79d99263a5a ve-TEST-01 -                opbwrtdacsm

Capability Flags:
o - Other; p - Repeater; b - Bridge; w - WLAN Access Point; r - Router
t - Telephone; d - DOCSIS cable device; a - Station; c - Customer VLAN
s - Service VLAN; m - Two-port MAC Relay (TPMR)

1 neighbor(s) listed.
```

`networkd` itself sends LLDP packets with the "Router" and "Station" capabilities enabled so it should probably be possible to add a test to make sure they are present.

Reply:
FWIW It would be great to fuzz it too. To judge from https://storage.googleapis.com/oss-fuzz-coverage/systemd/reports/20240301/linux/src/systemd/src/libsystemd-network/lldp-neighbor.c.html#L764 it isn't covered.

Reply:
Thanks. Added more test cases, and extended the fuzzer.

Reply:
Thanks! I think something like
```c
        assert_se(f = memstream_init(&m));
        (void) json_variant_dump(v, JSON_FORMAT_PRETTY|JSON_FORMAT_COLOR|JSON_FORMAT_SOURCE, f, NULL);
```
can be added to the fuzz target (mostly to help MSan make sure that all the fields are in good shape) but it can be done later in a separate PR probably.

Reply:
I thought that, but I do not want to slow down the process.

Reply:
I think we should do it… More coverage is always better. But it can be done separately.

Reply:
> I do not want to slow down the process

I don't think it should slow it down significantly. The fuzz target processes one LLDP packet at a time and there are no deeply nested structures there or anything like that. I think it would be a bug if a bunch of flat TLVs (whose total length is limited at the top) could cause the fuzz target to time out. I'll open a PR.